### PR TITLE
chore: pin pytest version

### PR DESCRIPTION
**Pull request by tswast**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-bigquery-dataframes/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> 🦕

### CODE REVIEW

**Review by tswast — COMMENTED** [review]
set max version instead of exact pin

**Review by GarrettWu — APPROVED** [review]

File: noxfile.py [hunk 1]
```diff
@@ -30,6 +30,7 @@
 
 BLACK_VERSION = "black==22.3.0"
 ISORT_VERSION = "isort==5.12.0"
+PYTEST_VERSION = "pytest<=7.4.4"
```
Comment (parthea) [inline noxfile.py:33]: nit: add a comment to capture the specific reason for adding an upper limit to `pytest` (with a link to the corresponding issue to follow up) . If there was a bad release of pytest, consider excluding the specific bad release.

**Review by tswast — COMMENTED** [review]
comment why we set max version

File: noxfile.py (hunk 1 above)
Comment (tswast) [reply noxfile.py]: Added! As far as I can tell from recent test failures it's that https://github.com/str0zzapreti/pytest-retry isn't compatible with 8.x, which was released this weekend.

File: noxfile.py [hunk 2]
```diff
@@ -30,6 +30,9 @@
 
 BLACK_VERSION = "black==22.3.0"
 ISORT_VERSION = "isort==5.12.0"
+
+# pytest-retry is not yet compatible with python 8.x.
```
Comment (parthea) [inline noxfile.py:34]:
```suggestion
# pytest-retry is not yet compatible with IPython 8.x.
```

Comment (parthea) [reply noxfile.py:34]: Please could you add a link to the upstream issue if it exists?

Comment (tswast) [reply noxfile.py]: Filed and linked here, thanks: https://github.com/str0zzapreti/pytest-retry/issues/32

**Review by parthea — APPROVED** [review]

**Pull request merged**